### PR TITLE
fix double notification - fixed #5702

### LINKED ISSUE #5702: Double notification for a same user an same event
You have notification defined in root entity as recursive.
You have notification defined in sub-entities.

In 9.1.x notification was sent only for notification from sub-entity.
Since 9.2.x notification was sent for notification from sub-entity + root entity.

Example : 
- notification for create ticket defined in root entity as recursive with a template and requester as recipient
- notification for create ticket defined in sub-entity with specific template and requester as recipient
=> the recipient receives 2 different notifications for the same event 

Bug introduce by: https://github.com/glpi-project/glpi/commit/5b5485e5e37f74afa531266134ded3cdccea6998#diff-b0744d216484b212b1e102240973bc57

### PR DESCRIPTION
**Pull request by yllen**
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #5702

### CODE REVIEW

File: inc/notificationeventabstract.class.php [hunk 1]
```diff
@@ -62,11 +62,10 @@ static public function raise(
       global $CFG_GLPI;
       if ($CFG_GLPI['notifications_' . $options['mode']]) {
          $entity = $notificationtarget->getEntity();
+         $processed = [];
```
Comment (trasher) [inline inc/notificationeventabstract.class.php]: Just make sure `$proposed` is an array.

File: inc/notificationeventabstract.class.php [hunk 2]
```diff
@@ -62,11 +62,10 @@ static public function raise(
...
          if (isset($options['processed'])) {
-            $processed = &$options['processed'];
+            $processed = $options['processed'];
```
Comment (trasher) [inline inc/notificationeventabstract.class.php]: No need for `&` if `$options['processed']` is unset.

**Comment by trasher** [comment]
@yllen please view the commit I added and my comments on it https://github.com/glpi-project/glpi/pull/5706/commits/ccd65243ccb13ae3f8a2409294ff67f849373bf7

File: inc/notificationeventabstract.class.php (hunk 2 above)
Comment (yllen) [reply inc/notificationeventabstract.class.php]: Not agree with this change. Array was sent to raiseEvent as empty one.

Log with your change  =>  both notifications are sent

==> mail.log <==
2019-03-29 09:50:33 [8@nelly]
Un courriel pour glpi@ll-experts.com a été ajouté à la file d'attente : [[NELLY] #0000327] new groupe nelly


==> php-errors.log <==
[2019-03-29 09:50:33] glpiphplog.DEBUG: NotificationEvent::raiseEvent() in /home/yllen/git94/inc/notificationevent.class.php line 165
processed johann Array
  (
      [mailing] => Array
          (
          )
  
  )
   {"user":"8@nelly"} 

==> mail.log <==
2019-03-29 09:50:33 [8@nelly]
Un courriel pour glpi@ll-experts.com a été ajouté à la file d'attente : [[NELLY] #0000327] Nouveau groupe Root

==> php-errors.log <==
[2019-03-29 09:50:33] glpiphplog.DEBUG: NotificationEvent::raiseEvent() in /home/yllen/git94/inc/notificationevent.class.php line 165
processed johann Array
  (
      [mailing] => Array
          (
          )
  
  )
   {"user":"8@nelly","mem_usage":"0.152\", 10.59Mio)"} 


Log with my commit => one notification of sub entity is sent

==> mail.log <==
2019-03-29 09:53:15 [8@nelly]
Un courriel pour glpi@ll-experts.com a été ajouté à la file d'attente : [[NELLY] #0000327] new groupe nelly

==> php-errors.log <==
[2019-03-29 09:53:15] glpiphplog.DEBUG: NotificationEvent::raiseEvent() in /home/yllen/git94/inc/notificationevent.class.php line 165
processed yllen Array
  (
      [mailing] => Array
          (
              [fr_FR] => Array
                  (
                      [glpi@ll-experts.com] => Array
                          (
                              [language] => fr_FR
                              [additionnaloption] => Array
                                  (
                                      [show_private] => 1
                                      [usertype] => 1
                                  )
  
                              [username] => MOED nelly
                              [users_id] => 8
                              [email] => glpi@ll-experts.com
                          )
  
                  )
  
          )
  
  )
   {"user":"8@nelly"} 
[2019-03-29 09:53:15] glpiphplog.DEBUG: NotificationEvent::raiseEvent() in /home/yllen/git94/inc/notificationevent.class.php line 165
processed yllen Array
  (
      [mailing] => Array
          (
              [fr_FR] => Array
                  (
                      [glpi@ll-experts.com] => Array
                          (
                              [language] => fr_FR
                              [additionnaloption] => Array
                                  (
                                      [show_private] => 1
                                      [usertype] => 1
                                  )
  
                              [username] => MOED nelly
                              [users_id] => 8
                              [email] => glpi@ll-experts.com
                          )
  
                  )
  
          )
  
  )
   {"user":"8@nelly","mem_usage":"0.019\", 10.48Mio)"}

**Comment by trasher** [comment]
OK, I'll remove the commit and merge your original proposal. Tank you for the feedback :)

**Pull request merged**